frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx

```diff
@@ -24,7 +24,7 @@ const PipelineRunTabDetails: React.FC<PipelineRunTabDetailsProps> = ({
   workflowName,
 }) => {
   const { namespace, project } = usePipelinesAPI();
-
+  const epoch = new Date(0).toISOString().slice(0, -5) + 'Z';
```

Comment:
@christianvogt no worries , just to clarify or summarize the things , now I need to create a string constant value
(const INVALID_TIMESTAMP=1970-01-01T00:00:00Z)
in concepts/pipelines/const.ts ? and add the comments

Reply:
Is there a reason we can't just write a utility to say `isEmptyDateKF(resourceKF.finished_at)` and it would just isolate this logic? I find it will be more pragmatic if we just lock away this logic away and stop writing inline functionality. Make it a common utility for all of the DS Pipelines effort.

The implementation of this method can be comparing a static string or doing proper date conversions, it doesn't matter either way as it just needs to get the job done.

Reply:
Two solution under consideration
1) Using  the constant
2) Implementing the utility function
@andrewballantyne @christianvogt which one do I need to go with?

Reply:
Write a utility, use a constant or whatever you want inside it... stop making custom logic comparisons... if we can't use OOTB ISO format, don't write special code to do it. It's a kubeflow concept, write a utility for it.

`a === b` doesn't _need_ to be anything more than a comparison... `isoFormat.split.doOtherThing === b` is now custom logic that you should not ever rewrite, so put it into a utility.

DRY with some intentional effort here not to make a need to refactor.